unbound-full is broken on unstable

### Describe the bug
unbound-full cannot be built on the unstable channel
error message:

```
/nix/store/js66s0xwjnzg0ggi2lq9bcvlk6x2za13-binutils-2.35.2/bin/ld: /build/testbound.vQYPwg.ltrans11.ltrans.o: in function `dtio_output_cb':
/build/unbound-1.13.2/dnstap/dtstream.c:1488: undefined reference to `dtio_disable_brief_write'
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. nix build .#unbound-full

### Expected behavior
It should obviously build

### Notify maintainers
cc @ehmry @fpletz @globin

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: unbound-full
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/tools/networking/unbound/default.nix
+++ b/pkgs/tools/networking/unbound/default.nix
@@ -32,6 +32,9 @@
 , withDNSTAP ? false
 , withTFO ? false
 , withRedis ? false
+# Avoid .lib depending on openssl.out
+# The build gets a little hacky, so in some cases we disable this approach.
+, withSlimLib ? stdenv.isLinux && !stdenv.hostPlatform.isMusl && !withDNSTAP
 , libnghttp2
 }:
 
@@ -105,10 +108,9 @@ stdenv.mkDerivation rec {
       --prefix PATH : ${lib.makeBinPath [ openssl ]}
   '';
 
-  preFixup = lib.optionalString (stdenv.isLinux && !stdenv.hostPlatform.isMusl) # XXX: revisit
+  preFixup = lib.optionalString withSlimLib
     # Build libunbound again, but only against nettle instead of openssl.
     # This avoids gnutls.out -> unbound.lib -> openssl.out.
-    # There was some problem with this on Darwin; let's not complicate non-Linux.
     ''
       configureFlags="$configureFlags --with-nettle=${nettle.dev} --with-libunbound-only"
       configurePhase

